### ISSUE TITLE
iMac-20, mid 2009 (9,1) 9 beeps after Patcher Catalina

### ISSUE THREAD

**Issue by g-horakhty** [dosdude1/macos-catalina-patcher#218]
This EFI certainly got corrupted, during the installation, does anyone know if using the "firmware RestorationCD1.9.dmg" can solve this problem?

**Comment by dosdude1** [comment]
Catalina Patcher and Catalina Installation does not touch the EFI firmware in any way, shape, or form. Did you apply the "APFS ROM Patch" (a separate tool, NOT included with Catalina Patcher)? If so, that is what caused the issue, and the only way to recover is to reflash externally using an EEPROM programmer with a known-good dump from another system.

**Comment by g-horakhty** [comment]
Yes, I tried installing the "APFS ROM Patch", but will just resoldering and reflashing fix it? Thanks.

**Comment by 5jd** [comment]
> Yes, I tried installing the "APFS ROM Patch", but will just resoldering and reflashing fix it? Thanks.

Hi, i had the exact same issue but i've resolved it using the backup ROM that APFS ROM Patcher created, then i used a generic EEPROM programmer that i bought on Amazon using the provided clamp (not recommended but i had no soldering tools).
tl;dr: Don't use APFS ROM Patcher on a iMac9,1 especially with a Nvidia chipset. (it is deprecated anyways)
